src/commands/dockerInstallerBase.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import ChildProcessProvider from '../debugging/coreclr/ChildProcessProvider';
+import { LocalFileSystemProvider } from '../debugging/coreclr/fsProvider';
+import { OSTempFileProvider } from '../debugging/coreclr/tempFileProvider';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { streamToFile } from '../utils/httpRequest';
+import LocalOSProvider from '../utils/LocalOSProvider';
+
+export abstract class DockerInstallerBase {
+    protected abstract downloadUrl: string;
+    protected abstract fileExtension: string;
+    protected abstract installationMessage: string;
+    protected abstract getInstallCommand(fileName: string): string;
+
+    public async downloadAndInstallDocker(): Promise<void> {
+        const confirmInstall: string = localize('vscode-docker.commands.DockerInstallerBase.confirm', 'Are you sure you want to install docker on this machine.');
+        const installTitle: string = localize('vscode-docker.commands.DockerInstallerBase.install', 'Install');
+        const downloadingMessage: string = localize('vscode-docker.commands.DockerInstallerBase.downloading', 'Downloading Docker installer...');
+        const downloadCompleteMessage: string = localize('vscode-docker.commands.DockerInstallerBase.downloadCompleteMessage', 'Download completed');
+        let downlaodedFileName: string;
+
+        // no need to check result - cancel will throw a UserCancelledError
+        await ext.ui.showWarningMessage(confirmInstall, { modal: true }, { title: installTitle });
+
+        await vscode.window.withProgress(
+            { location: vscode.ProgressLocation.Notification, title: downloadingMessage },
+            async () => {
+                downlaodedFileName = await this.downloadInstaller();
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                vscode.window.showInformationMessage(downloadCompleteMessage);
```

Comment:
Do we need another toast to say download is complete? IMO when the "Downloading Docker installer..." progress toast disappears it's pretty clear that download is done. Also, immediately after this we show the installation message, so I'd rather just remove this toast entirely.

Reply:
@BigMorty, Your call.

Reply:
Users have repeatedly stated that toasts are annoying. We should minimize them.

Reply:
Agree, that the 'Downloaded' is unnecessary. Removed it now. Also sent email to get Kaitlin (UX) input, based on her input, I will adjust.